src/main/java/com/tterrag/k9/commands/CommandMappings.java

```diff
@@ -152,37 +159,45 @@ public void save(File dataFolder, Gson gson) {
         Flux<Mapping> ret = updateCheck.thenMany(mcver.flatMapMany(v -> type == null ? downloader.lookup(name, v) : downloader.lookup(type, name, v)));
         if (ctx.hasFlag(FLAG_CONVERT)) {
             String convertTo = ctx.getFlag(FLAG_CONVERT);
-            // TODO some way to look this up dynamically?
-            MappingDownloader<?, ?> type;
-            if (convertTo.toLowerCase(Locale.ROOT).startsWith("m")) {
-                type = McpDownloader.INSTANCE;
-            } else if (convertTo.toLowerCase(Locale.ROOT).startsWith("y")) {
-                type = YarnDownloader.INSTANCE;
-            } else {
+            CommandMappings<?> otherCommand = getOtherCommand(convertTo);
+            if (otherCommand == null)
                 return ctx.error("Unknown mapping type for conversion: " + convertTo).thenMany(Flux.empty());
-            }
-            Mono<? extends MappingDatabase<?>> dbCache = mcver.flatMap(type::getDatabase).cache();
+            Mono<? extends MappingDatabase<?>> dbCache = mcver.flatMap(otherCommand.downloader::getDatabase).cache();
             return ret
                     .flatMap(m -> dbCache.<Mapping>flatMap(db -> Mono.justOrEmpty(m.<Mapping>convert(db))));
         }
         return ret;
     }
 
+    @Nullable
+    private CommandMappings<?> getOtherCommand(String convertTo) {
```

Comment:
Just change the behavior to require the proper name, honestly should have just done that in the first place. The argument to -c should be whatever the "base" command name is, so "yarn", "mcp", etc. Not much point in using a map that's only looped over. This fuzzy matching seems like more trouble than its worth.

Reply:
Done